### ISSUE TITLE
Split C++ library from Python bindings

### ISSUE THREAD

**Issue by mfansler** [conda-forge/nlopt-feedstock#41]
### Comment:

This feedstock currently has a Python-centric recipe. It would make more sense to split this into a compiled library and a Python bindings package, so that other packages could depend on just the compiled library. For example, [the conda-forge/r-nloptr-feedstock](https://github.com/conda-forge/r-nloptr-feedstock) would benefit from this. Up to now that feedstock has been redistributing, but I am hesitant to switch to using the `nlopt` package as a dependency in its current form because it entails installing Python in an (otherwise strictly) R environment, something I believe is almost never necessary and overcomplicates solves.

Opening this Issue for discussion.

**Comment by IsabelParedes** [comment]
I agree it should be split. Adding another output to this feedstock should be relatively simple.

**Comment by IsabelParedes** [comment]
Or maybe creating another feedstock for `libnlopt` could be an easier option, and no changes would be needed here.
